Model/CustomItemExportSchedulerModel.php

```diff
@@ -0,0 +1,294 @@
+
+                foreach ($customFields as $customField) {
+                    $fieldValue = $listData->getFields($customItem->getId())[$customField->getId()]->getValue();
+
+                    switch ($customField->getType()) {
+                        case 'date':
+                            $value = $fieldValue instanceof \DateTimeInterface ? $fieldValue->format('Y-m-d') : $fieldValue;
+                            break;
+
+                        case 'datetime':
+                            $value = $fieldValue instanceof \DateTimeInterface ? $fieldValue->format('Y-m-d H:i:s') : $fieldValue;
+                            break;
+
+                        case 'multiselect': $value = is_array($fieldValue) ? implode(',', $fieldValue) : $fieldValue;
+                            break;
+
+                            default: $value = $fieldValue;
+                    }
+
+                    $rowData[] = $value;
+                }
+
+                $fetchResult     = true;
+                $savedRow        = $rowData;
+                $contactOffset   = 0;
+                $customItemAdded = false;
+
+                while ($fetchResult) {
+                    $results = $this->getContactIds($customItem, self::CONTACT_LIMIT, $contactOffset);
+
+                    if (0 == count($results) && $customItemAdded) {
+                        break;
+                    }
+
+                    if (count($results) < self::CONTACT_LIMIT) {
+                        $fetchResult = false;
+                    } else {
+                        $contactOffset += self::CONTACT_LIMIT;
+                    }
+
+                    $rowData   = $savedRow;
+                    $rowData[] = implode(',', $results);
+
+                    if ($this->eventDispatcher->hasListeners(CustomItemEvents::ON_PROCESSING_FILE)) {
+                        $this->eventDispatcher->dispatch(CustomItemEvents::ON_PROCESSING_FILE);
+                    }
+
+                    fputcsv($handler, $rowData);
+                    $customItemAdded = true;
+                }
+            }
+        }
+
+        fclose($handler);
+    }
+
+    /**
+     * @param array<CustomField> $customFields
+     */
+    private function addExportFileHeaderToCsvFile(array $customFields): void
+    {
+        $header  = $this->getCSVHeader($customFields);
+        $handler = @fopen($this->filePath, 'ab+');
+        fputcsv($handler, $header);
+        fclose($handler);
+    }
+
+    /**
+     * @return array<int>
+     */
+    private function getContactIds(CustomItem $customItem, int $limit = 200, int $offset = 0): array
+    {
+        $contactIds = $this->customItemXrefContactRepository
+            ->getContactIdsLinkedToCustomItem($customItem->getId(), $limit, $offset);
+
+        return array_column($contactIds, 'contact_id');
+    }
+
+    public function sendEmail(CustomItemExportScheduler $customItemExportScheduler, string $filePath): void
```

Comment:
Sending emails should not be part of the model. This should be extracted into its own class/service.